reader_concurrency_semaphore.cc

```diff
@@ -1073,7 +1073,18 @@ future<> reader_concurrency_semaphore::do_wait_admission(reader_permit permit, r
         _execution_loop_future.emplace(execution_loop());
     }
 
-    const auto admit = can_admit_read(permit).decision;
+    uint64_t* stats[] = {
+        nullptr, // all_ok
+        &_stats.reads_queued_because_ready_list,
+        &_stats.reads_queued_because_used_permits,
+        &_stats.reads_queued_because_memory_resources,
+        &_stats.reads_queued_because_count_resources
```

Comment:
Better to use a static table with pointer-to-members, to avoid initializing it each time. Also, instead of nullptr, point to an unused field so you can skip the if.

Another option is to use a small array indexed by the reason.

Yet another option is to update the correct stat in can_admit_read() (requires mutable but that's legitimate for stats).

Reply:
Ah, I just now understood what you meant by static table with pointer-to-members. I will look into that, that would allow us to avoid the per-instance table.